python/fastquant/strategies/base.py

```diff
@@ -289,7 +294,33 @@ def next(self):
                         self.sell(exectype=bt.Order.StopTrail, trailpercent=self.stop_trail, size=final_size)
 
         elif self.sell_signal():
-            if stock_value > 0:
+            if self.allow_short == True:
+
+                # Sell short based on the closing price of the previous day
+                if self.execution_type == "close":
+
+                    if self.transaction_logging:
+                        self.log("SELL CREATE, %.2f" % self.dataclose[0])
+
+                    sell_prop_size = int(SELL_PROP * 
+                                         self.broker.getvalue() / 
+                                         self.dataclose[0])
+                    self.order = self.sell(size=sell_prop_size)
+                
+                 # Buy based on the opening price of the next closing day (only works "open" data exists in the dataset)
+                else:
+
+                    if self.transaction_logging:
+                        self.log("SELL CREATE, %.2f" % self.dataclose[1])
+                    
+                    sell_prop_size = int(self.short_max *
+                                         self.cash /
+                                         self.dataclose[1] /
+                                         (1 + self.commission + 0.001))
+                    self.order = self.sell(size=min(sell_prop_size, self.init_cash * SHORT_MAX))
```

Comment:
```suggestion
                    # The max incremental short allowed is the short that would lead to a cumulative short position
                    # equal to the maximum short position (initial cash times the maximum short ratio, which is 1.5 by default)
                    max_position_size = max((self.init_cash * SHORT_MAX / self.dataopen[1]) + self.position.size, 0)
                    self.order = self.sell(size=min(sell_prop_size, max_position_size))
```